src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -0,0 +1,244 @@
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          countryUI['ui:disabled'] = true;
+          countryFormData.countryName = USA.name;
+          return {
+            enum: [USA.name],
+            default: USA.name,
+          };
+        }
+        countryUI['ui:disabled'] = false;
+        return {
+          type: 'string',
+          enum: countries.map(country => country.label),
+        };
+      },
+    },
+  },
+  addressLine1: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:title': 'Street address',
+    'ui:errorMessages': {
+      required: 'Street address is required',
+      pattern: 'Street address must be under 100 characters',
+    },
+  },
+  addressLine2: {
+    'ui:title': 'Street address',
+  },
+  addressLine3: {
+    'ui:title': 'Street address',
+  },
+  city: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:errorMessages': {
+      required: 'City is required',
+      pattern: 'City must be under 100 characters',
+    },
+    'ui:options': {
+      replaceSchema: formData => {
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase'] === true) {
+          return {
+            type: 'string',
+            title: 'APO/FPO/DPO',
+            enum: militaryCities,
+          };
+        }
+        return {
+          type: 'string',
+          title: 'City',
+          minLength: 1,
+          maxLength: 100,
+          pattern: '^.*\\S.*',
+        };
+      },
+    },
+  },
+  stateCode: {
+    'ui:required': formData =>
+      formData[`${schemaKey}`].countryName === USA.name ||
+      formData[`${schemaKey}`]['view:livesOnMilitaryBase'],
+    'ui:title': 'State',
+    'ui:errorMessages': {
+      required: 'State is required',
+    },
+    'ui:options': {
+      hideIf: formData => {
+        // Because we have to update countryName manually in formData above,
+        // We have to check this when a user selects a non-US country and then selects
+        // the military base checkbox.
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          return false;
+        }
+        return formData[`${schemaKey}`].countryName !== USA.name;
+      },
+      updateSchema: formData => {
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          return {
+            enum: Object.keys(MILITARY_STATES),
+            enumNames: Object.values(MILITARY_STATES),
+          };
+        }
+        return {
+          enum: states50AndDC.map(state => state.value),
+          enumNames: states50AndDC.map(state => state.label),
+        };
+      },
+    },
+  },
+  province: {
+    'ui:title': 'State/Province/Region',
+    'ui:options': {
+      hideIf: formData => {
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          return true;
+        }
+        return formData[`${schemaKey}`].countryName === USA.name;
```

Comment:
Same situation here regarding the case of selecting a non-US country and then checking the box for military base.